import neccessary library

In [1]:
import numpy as np
import scipy.stats as stats
import seaborn as sns

import the dataset

In [5]:
tips = sns.load_dataset("tips")
tips.head()

total_bill   tip     sex smoker  day    time  size
0       16.99  1.01  Female     No  Sun  Dinner     2
1       10.34  1.66    Male     No  Sun  Dinner     3
2       21.01  3.50    Male     No  Sun  Dinner     3
3       23.68  3.31    Male     No  Sun  Dinner     2
4       24.59  3.61  Female     No  Sun  Dinner     4

define hypothesis

we want to test whether there is a avrage difference between the tips between male and female customers
1. null hypothesis(Ho): the average tips given by male are equal to the average tips given by female female
2. Alternative hypothesis(H1): the average tips given by the male are differnet than the average tips given by the female

since we are comparing the mean between the two independent groups (male and female), we will use an independent t-test.  

filter the tips by gender

In [7]:
male_tips = tips[tips["sex"] == "Male"]["tip"]
female_tips = tips[tips["sex"] == "Female"]["tip"]

perform the t test 
use scipy.stats.ttest_ind() to conduct the t-test

In [15]:
t_stat, p_val = stats.ttest_ind(male_tips, female_tips)
print("t-statistics = ", t_stat)
print("p-value = ", p_val)

t-statistics =  1.387859705421269
p-value =  0.16645623503456755


interpritation of result:

1. P value = If the p-value is less than the significance level (typically 0.05), we reject the null hypothesis
             this suggest a statistically significant difference in average tips between male and female customer.
2. t-statistics = it provide the measure of size of differnce provided in the sample data 

Interpretation:
Since the p-value (0.166) is greater than the typical significance level of 0.05, 
you fail to reject the null hypothesis. This indicates that there is no statistically significant 
difference in average tip amounts between male and female customers in this dataset.

What This Means:
In practical terms, this result suggests that, based on this data, we do not have enough 
evidence to conclude that the average tips given by male and female customers differ significantly. 
Any observed difference in the sample could be due to random chance rather than a true difference in the population.

how do we know the significant level :
In hypothesis testing, the significance level (α) is generally 
chosen before conducting the test, based on standard practices or the specific requirements of the analysis. 
The significance level represents the threshold at which we decide whether to reject the null hypothesis.

By convention, 0.05 is commonly used as the significance level,
meaning there’s a 5% risk of rejecting the null hypothesis when it's actually true. 
However, you can choose different values for 
α depending on how strict you want the test to be.

Adding Significance Level (α) to Your Code
If you'd like to explicitly include 
α in your code, you can define it at the beginning. 
This approach is useful for making the significance level clear and adjustable if you want to test different thresholds.

complete code = 

In [17]:
import seaborn as sns
import scipy.stats as stats

# Load dataset
tips = sns.load_dataset("tips")

# Separate data by gender
male_tips = tips[tips['sex'] == 'Male']['tip']
female_tips = tips[tips['sex'] == 'Female']['tip']

# Perform t-test
t_stat, p_val = stats.ttest_ind(male_tips, female_tips)

# Output the results
print("t-statistic:", t_stat)
print("p-value:", p_val)

# Interpret the result
if p_val < 0.05:
    print("Reject the null hypothesis: There is a significant difference in average tips between genders.")
else:
    print("Fail to reject the null hypothesis: No significant difference in average tips between genders.")


t-statistic: 1.387859705421269
p-value: 0.16645623503456755
Fail to reject the null hypothesis: No significant difference in average tips between genders.
